roles/lib_openshift/library/oc_project.py

```diff
@@ -1543,19 +1543,20 @@ def needs_update(self):
     def run_ansible(params, check_mode):
         '''run the idempotent ansible code'''
 
-        _ns = None
-        if params['node_selector'] is not None:
-            _ns = ','.join(params['node_selector'])
-
-        pconfig = ProjectConfig(params['name'],
-                                'None',
-                                params['kubeconfig'],
-                                {'admin': {'value': params['admin'], 'include': True},
-                                 'admin_role': {'value': params['admin_role'], 'include': True},
-                                 'description': {'value': params['description'], 'include': True},
-                                 'display_name': {'value': params['display_name'], 'include': True},
-                                 'node_selector': {'value': _ns, 'include': True},
-                                })
+        node_selector = ','.join(params.get('node_selector', [])) or None
```

Comment:
You're right, thanks for catching it. I've updated the patch.

I saw this gets used later on in `ProjectConfig.stringify` and there we only test the truthiness of the value, whereas in `OCProject.needs_update` we test whether the value is `None`.

Let's step back and keep the original 3 lines to avoid introducing any unintended bug.